examples/agent_client/examples/agent_client.rs

```diff
@@ -0,0 +1,324 @@
+use uuid::Uuid;
+
+///
+/// The sample usage of talos agent library
+///
+
+const BATCH_SIZE: i32 = 30_000;
+const TALOS_TYPE: TalosType = TalosType::InProcessMock;
+const PROGRESS_EVERY: i32 = 50_000;
+const NANO_IN_SEC: i32 = 1_000_000_000;
+const TARGET_RATE: f64 = 5_000_f64;
+const BASE_DELAY: Duration = Duration::from_nanos((NANO_IN_SEC as f64 / TARGET_RATE) as u64);
+
+fn make_configs() -> (AgentConfig, KafkaConfig) {
+    let cohort = "HostForTesting";
+    let agent = format!("agent-for-{}-{}", cohort, Uuid::new_v4());
+    let cfg_agent = AgentConfig {
+        agent: agent.clone(),
+        cohort: cohort.to_string(),
+        buffer_size: 10_000,
+    };
+
+    let cfg_kafka = KafkaConfig {
+        brokers: "127.0.0.1:9093".to_string(),
+        group_id: agent,
+        enqueue_timeout_ms: 10,
+        message_timeout_ms: 15000,
+        fetch_wait_max_ms: 6000,
+        certification_topic: "dev.ksp.certification".to_string(),
+        log_level: RDKafkaLogLevel::Info,
+        talos_type: TALOS_TYPE,
+    };
+
+    (cfg_agent, cfg_kafka)
+}
+
+fn make_candidate(xid: String) -> CertificationRequest {
+    let tx_data = CandidateData {
+        xid,
+        readset: Vec::new(),
+        readvers: Vec::new(),
+        snapshot: 5,
+        writeset: Vec::from(["3".to_string()]),
+    };
+
+    CertificationRequest {
+        message_key: "12345".to_string(),
+        candidate: tx_data,
+    }
+}
+
+fn name_rate(v: Duration) -> String {
+    let nr = 1_f64 / v.as_secs_f64();
+    if nr > 1000_f64 {
+        format!("{}k/s", nr / 1000_f64)
+    } else {
+        format!("{}/s", nr)
+    }
+}
+
+// async fn make_agent() -> Box<TalosAgentType> {
+//     let (cfg_agent, cfg_kafka) = make_configs();
+//
+//     TalosAgentBuilder::new(cfg_agent)
+//         .with_kafka(cfg_kafka)
+//         .build()
+//         .unwrap_or_else(|e| panic!("{}", format!("Unable to build agent {}", e)))
+// }
+
+async fn make_agentv2(publish_times: &Arc<Mutex<HashMap<String, u64>>>) -> Box<TalosAgentType> {
+    let (cfg_agent, cfg_kafka) = make_configs();
+
+    TalosAgentBuilder::new(cfg_agent)
+        .with_kafka(cfg_kafka)
+        .build_v2(publish_times)
+        .await
+        .unwrap_or_else(|e| panic!("{}", format!("Unable to build agent {}", e)))
+}
+
+#[tokio::main]
+async fn main() -> Result<(), String> {
+    env_logger::init();
+
+    thread::sleep(Duration::from_secs(5));
+
+    certify(BATCH_SIZE).await
+}
+
+async fn certify(batch_size: i32) -> Result<(), String> {
+    info!("Certifying {} transactions", batch_size);
+
+    // "global" state where we will collect publishing times: A - B, where
+    // B: transaction start time
+    // A: the end of call to kafka send_message(candidate)
+    let publish_times: Arc<Mutex<HashMap<String, u64>>> = Arc::new(Mutex::new(HashMap::new()));
+    let mut tasks = Vec::<JoinHandle<Result<Timing, String>>>::new();
+    let agent = Arc::new(make_agentv2(&publish_times).await);
```

Comment:
Just use `Arc::clone(publish_times)` rather than passing the reference of the clone all the way down, and using clone in the `new` method of `StateManager`

Reply:
Thanks. This is fixed now.